tensorbay/opendataset/SegTrack/loader.py

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env python3
+#
+# Copytright 2021 Graviti. Licensed under MIT License.
+#
+# pylint: disable=invalid-name
+# pylint: disable=missing-module-docstring
+
+import os
+from typing import Callable
+
+import numpy as np
+
+from ...dataset import Data, Dataset
+from ...label import SemanticMask
+from .._utility import glob
+
+try:
+    from PIL import Image
+except ModuleNotFoundError:
+    from .._utility.mocker import Image
+
+DATASET_NAME = "SegTrack"
+_SEGMENT_NAMES = ("birdfall2", "cheetah", "girl", "monkeydog", "parachute", "penguin")
+
+
+def SegTrack(path: str) -> Dataset:
+    """Dataloader of the `SegTrack`_ dataset.
+
+    .. _SegTrack: http://cpl.cc.gatech.edu/projects/SegTrack/
+
+    The file structure of SegTrack looks like::
+
+        <path>
+            birdfall2/
+                birdfall2_00018.png
+                ...
+                ground-truth/
+                    birdfall2_00018.png
+                    ...
+            cheetah/
+                chasedeer_frame_0001.bmp
+                ...
+                ground-truth/
+                    chasedeer_00000.png
+                    ...
+            girl/
+                5117-8_70161.bmp
+                ...
+                ground-truth/
+                    0.bmp
+                    ...
+            monkeydog/
+                195.bmp
+                ...
+                ground-truth/
+                    Comp_00195.png
+                    ...
+            parachute/
+                parachute_00000.png
+                ...
+                ground-truth/
+                    parachute_00000.png
+                    ...
+            penguin/
+                penguin_00000.bmp
+                ...
+                ground_truth/
+                    penguin_00000.png
+                    ...
+
+    Arguments:
+        path: The root directory of the dataset.
+
+    Returns:
+        Loaded :class:`~tensorbay.dataset.dataset.Dataset` instance.
+
+    """
+    root_path = os.path.join(os.path.abspath(os.path.expanduser(path)))
+    dataset = Dataset(DATASET_NAME)
+    dataset.load_catalog(os.path.join(os.path.dirname(__file__), "catalog.json"))
+
+    for segment_name in _SEGMENT_NAMES:
+        segment = dataset.create_segment(segment_name)
+        reformat_filename = _get_reformat_func(segment_name)
+        segment_directory = os.path.join(root_path, segment_name)
+        mask_directory = os.path.join(segment_directory, "masks")
+        os.makedirs(mask_directory, exist_ok=True)
+        mask_name = "ground_truth" if segment_name == "penguin" else "ground-truth"
+        original_mask_directory = os.path.join(segment_directory, mask_name)
+        for image_path in glob(os.path.join(segment_directory, "*.*")):
+            data = Data(image_path)
+            filename = os.path.splitext(os.path.basename(image_path))[0]
+            mask_path = os.path.join(mask_directory, f"{filename}.png")
+            if not os.path.exists(mask_path):
```

Comment:
Why do we need this judgment instead of just replacing it with a new png

Reply:
I think modifying original mask when loading dataset is not good